src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,21 @@
+import {Component, Input, OnInit, TemplateRef, ViewContainerRef} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent implements OnInit {
+    @Input() template: TemplateRef<any> | null = null;
+
+    constructor(private readonly viewContainerRef: ViewContainerRef) {}
+
+    ngOnInit(): void {
+        if (this.template) {
+            this.viewContainerRef.clear();
+            this.viewContainerRef.createEmbeddedView(this.template);
+        } else {
+            this.viewContainerRef.clear();
+        }
+    }
```

Comment:
Увидел сперва `*ngTemplateOutlet` - понравилось, хоть и не изучали еще, но сейчас я немного в смятении)) - а зачем нам `ngOnInit` и `viewContainerRef`? И валидно ли для такой ситуации использовать `ngOnInit`? Инпут же у нас динамически меняется. В общем, хотелось бы понять ход твоих мыслей, чтобы убедиться, что материал точно усвоен верно.

И по существующему коду, я бы вынес все таки общую часть и не использовал if/else:
```suggestion
    ngOnInit(): void {
        this.viewContainerRef.clear();
        
        if (this.template) {
            this.viewContainerRef.createEmbeddedView(this.template);
        }
    }
```